### PR TITLE
fixed guava cve

### PR DESCRIPTION
**Pull request by conker84**
Fixes `Security vulnerability with Neo4j enterprise 4.2.7` trello card

https://cve.mitre.org/cgi-bin/cvekey.cgi?keyword=CVE-2018-10237

`Unbounded memory allocation in Google Guava 11.0 through 24.x before 24.1.1`

### CODE REVIEW

**Comment by conker84** [comment]
It's the flaky test that makes the build red `apoc.util.UtilIT > redirectShouldWorkWhenProtocolNotChangesWithUrlLocation`

File: full/src/test/java/apoc/util/HdfsTestUtils.java [hunk 1]
```diff
@@ -31,16 +33,23 @@ private static String getHadoopHome() {
...
     	conf.set("fs.defaultFS", "hdfs://localhost");
 		File hdfsPath = new File(System.getProperty("user.dir") + File.separator + "hadoop" + File.separator + "hdfs");
-		conf.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR, hdfsPath.getAbsolutePath());
+        hdfsPath.setWritable(true);
+        conf.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR, hdfsPath.getAbsolutePath());
 		MiniDFSCluster miniDFSCluster = new MiniDFSCluster.Builder(conf)
-                .nameNodePort(12345)
-                .nameNodeHttpPort(12341)
+                .nameNodePort(getFreePort())
+//                .nameNodeHttpPort(12341)
```
Comment (fbiville) [inline full/src/test/java/apoc/util/HdfsTestUtils.java:52]: is it meant to be removed or was it temporary?

File: core/build.gradle [hunk 2]
```diff
@@ -110,8 +110,8 @@ dependencies {
 
     testCompile group: 'org.apache.hive', name: 'hive-jdbc', version: '1.2.2', withoutServers
 
-    compileOnly group: 'org.apache.hadoop', name: 'hadoop-hdfs', version: '2.7.5', withoutServers
-    compileOnly group: 'org.apache.hadoop', name: 'hadoop-common', version: '2.7.5', withoutServers
+    compileOnly group: 'org.apache.hadoop', name: 'hadoop-hdfs', version: '3.3.1', withoutServers
+    compileOnly group: 'org.apache.hadoop', name: 'hadoop-common', version: '3.3.1', withoutServers
 
     compile group: 'org.apache.commons', name: 'commons-math3', version: '3.6.1'
     // explicit update comomns.io version
```
Comment (fbiville) [inline core/build.gradle:117]:
```suggestion
    // explicit update commons.io version
```

**Review by fbiville — APPROVED** [review]

File: full/src/test/java/apoc/util/HdfsTestUtils.java (hunk 1 above)
Comment (conker84) [reply full/src/test/java/apoc/util/HdfsTestUtils.java:52]: removed

File: core/build.gradle (hunk 2 above)
Comment (conker84) [reply core/build.gradle:117]: done

**Pull request merged**